### PR TITLE
Add version: GoLang.Go version 1.20.4

### PR DESCRIPTION
**Pull request by kkamegawa**
- [ ] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [ ] This PR only modifies one (1) manifest
- [ ] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [ ] Have you tested your manifest locally with `winget install --manifest manifests\g\GoLang\Go\1.20.4`?
- [ ] Does your manifest conform to the [1.4 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.4.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----

 ###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/104907)

### CODE REVIEW

File: manifests/g/GoLang/Go/1.20.4/GoLang.Go.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+- interactive
+- silent
+- silentWithProgress
+UpgradeBehavior: install
+Commands:
+- go
+FileExtensions:
+- go
+- gohtml
+Installers:
```
Comment (russellbanks) [inline manifests/g/GoLang/Go/1.20.4/GoLang.Go.installer.yaml:18]:
```suggestion
ReleaseDate: 2023-05-02
Installers:
```

File: manifests/g/GoLang/Go/1.20.4/GoLang.Go.installer.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
+# Created using wingetcreate 1.2.5.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.4.0.schema.json
+
+PackageIdentifier: GoLang.Go
+PackageVersion: 1.20.4
+InstallerLocale: en-US
```
Comment (russellbanks) [inline manifests/g/GoLang/Go/1.20.4/GoLang.Go.installer.yaml:6]:
```suggestion
InstallerLocale: en-US
InstallerType: wix
```

File: manifests/g/GoLang/Go/1.20.4/GoLang.Go.installer.yaml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+- silentWithProgress
+UpgradeBehavior: install
+Commands:
+- go
+FileExtensions:
+- go
+- gohtml
+Installers:
+- Architecture: x86
+  InstallerType: wix
```
Comment (russellbanks) [inline manifests/g/GoLang/Go/1.20.4/GoLang.Go.installer.yaml:20]:
```suggestion
```

File: manifests/g/GoLang/Go/1.20.4/GoLang.Go.installer.yaml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  InstallerUrl: https://go.dev/dl/go1.20.4.windows-386.msi
```
Comment (russellbanks) [inline manifests/g/GoLang/Go/1.20.4/GoLang.Go.installer.yaml:21]:
```suggestion
  InstallerUrl: https://dl.google.com/go/go1.20.4.windows-386.msi
```

File: manifests/g/GoLang/Go/1.20.4/GoLang.Go.installer.yaml [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+  InstallerSha256: 03BDB95782028AF5140233DE1649D5E1B76B046955502F9C51FC364DC1AC4C5C
+  ProductCode: '{7697003C-D4E0-4007-8A23-8296D9F87917}'
+  AppsAndFeaturesEntries:
+  - DisplayName: Go Programming Language 386 go1.20.3
```
Comment (russellbanks) [inline manifests/g/GoLang/Go/1.20.4/GoLang.Go.installer.yaml:25]:
```suggestion
  - DisplayName: Go Programming Language 386 go1.20.4
```

File: manifests/g/GoLang/Go/1.20.4/GoLang.Go.installer.yaml [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    UpgradeCode: '{1C3114EA-08C3-11E1-9095-7FCA4824019B}'
+- Architecture: x64
+  InstallerType: wix
```
Comment (russellbanks) [inline manifests/g/GoLang/Go/1.20.4/GoLang.Go.installer.yaml:28]:
```suggestion
```

File: manifests/g/GoLang/Go/1.20.4/GoLang.Go.installer.yaml [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+  InstallerUrl: https://go.dev/dl/go1.20.4.windows-amd64.msi
+  InstallerSha256: D9F24142072EC50B06325B317B9F40E7C50B7A22949F55B39B467718193C6E46
+  ProductCode: '{01C7C71B-B12F-4146-9A54-41530E7CB60B}'
+  AppsAndFeaturesEntries:
+  - DisplayName: Go Programming Language amd64 go1.20.3
+    UpgradeCode: '{22EA7650-4AC6-4001-BF29-F4B8775DB1C0}'
+- Architecture: arm64
+  InstallerType: wix
```
Comment (russellbanks) [inline manifests/g/GoLang/Go/1.20.4/GoLang.Go.installer.yaml:36]:
```suggestion
```

File: manifests/g/GoLang/Go/1.20.4/GoLang.Go.installer.yaml [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+  InstallerUrl: https://go.dev/dl/go1.20.4.windows-386.msi
+  InstallerSha256: 03BDB95782028AF5140233DE1649D5E1B76B046955502F9C51FC364DC1AC4C5C
+  ProductCode: '{7697003C-D4E0-4007-8A23-8296D9F87917}'
+  AppsAndFeaturesEntries:
+  - DisplayName: Go Programming Language 386 go1.20.3
+    UpgradeCode: '{1C3114EA-08C3-11E1-9095-7FCA4824019B}'
+- Architecture: x64
+  InstallerType: wix
+  InstallerUrl: https://go.dev/dl/go1.20.4.windows-amd64.msi
```
Comment (russellbanks) [inline manifests/g/GoLang/Go/1.20.4/GoLang.Go.installer.yaml:29]:
```suggestion
  InstallerUrl: https://dl.google.com/go/go1.20.4.windows-amd64.msi
```

File: manifests/g/GoLang/Go/1.20.4/GoLang.Go.installer.yaml [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+  InstallerSha256: D9F24142072EC50B06325B317B9F40E7C50B7A22949F55B39B467718193C6E46
+  ProductCode: '{01C7C71B-B12F-4146-9A54-41530E7CB60B}'
+  AppsAndFeaturesEntries:
+  - DisplayName: Go Programming Language amd64 go1.20.3
+    UpgradeCode: '{22EA7650-4AC6-4001-BF29-F4B8775DB1C0}'
+- Architecture: arm64
+  InstallerType: wix
+  InstallerUrl: https://go.dev/dl/go1.20.4.windows-arm64.msi
```
Comment (russellbanks) [inline manifests/g/GoLang/Go/1.20.4/GoLang.Go.installer.yaml:37]:
```suggestion
  InstallerUrl: https://dl.google.com/go/go1.20.4.windows-arm64.msi
```

File: manifests/g/GoLang/Go/1.20.4/GoLang.Go.installer.yaml [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+  InstallerSha256: B9F53616FC81ADE97999AE9117ACE8A0A7D2785F60436DF4F3283D6153A38153
+  ProductCode: '{DAEAAB0E-B141-4656-9788-5C5B7C8E2CE0}'
+  AppsAndFeaturesEntries:
+  - DisplayName: Go Programming Language arm64 go1.20.3
```
Comment (russellbanks) [inline manifests/g/GoLang/Go/1.20.4/GoLang.Go.installer.yaml:41]:
```suggestion
  - DisplayName: Go Programming Language arm64 go1.20.4
```

**Review by russellbanks — CHANGES_REQUESTED** [review]

File: manifests/g/GoLang/Go/1.20.4/GoLang.Go.installer.yaml [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+  AppsAndFeaturesEntries:
+  - DisplayName: Go Programming Language 386 go1.20.3
+    UpgradeCode: '{1C3114EA-08C3-11E1-9095-7FCA4824019B}'
+- Architecture: x64
+  InstallerType: wix
+  InstallerUrl: https://go.dev/dl/go1.20.4.windows-amd64.msi
+  InstallerSha256: D9F24142072EC50B06325B317B9F40E7C50B7A22949F55B39B467718193C6E46
+  ProductCode: '{01C7C71B-B12F-4146-9A54-41530E7CB60B}'
+  AppsAndFeaturesEntries:
+  - DisplayName: Go Programming Language amd64 go1.20.3
```
Comment (russellbanks) [inline manifests/g/GoLang/Go/1.20.4/GoLang.Go.installer.yaml:33]:
```suggestion
  - DisplayName: Go Programming Language amd64 go1.20.4
```

**Review by russellbanks — CHANGES_REQUESTED** [review]

**Review by russellbanks — CHANGES_REQUESTED** [review]
`InstallerType: wix` still needs to be removed from installer level as having it as root level covers it for all installers

**Review by ItzLevvie — APPROVED** [review]

**Pull request merged**